loading scalar.ts not working on node@20

In Node.js v20, loading TypeScript schema entry not working as in [examples/nextjs](https://github.com/uhyo/nitrogql/tree/master/examples/nextjs).

- node version: 20.8.1
- nitrogql version:
  - `@nitrogql/cli`: 1.3.0
  - `@nitrogql/graphql-loader`: 1.3.0

<details>

<summary>error on minimal reproduction: examples/nextjs</summary>

```
> nextjs@1.3.0 generate
> nitrogql generate

/Users/futaiwata/nitrogql/examples/nextjs/src/schema/scalar.ts:1
import { GraphQLSchema } from "graphql";
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at internalCompileFunction (node:internal/vm:73:18)
    at loadCJSModule (node:internal/modules/esm/translators:155:23)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:234:7)
    at ModuleJob.run (node:internal/modules/esm/module_job:217:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
    at async loadSchemaJs (file:///Users/futaiwata/nitrogql/examples/nextjs/node_modules/@nitrogql/core/dist/loader.js:5:18)

Node.js v20.8.1
Error: Command failed: node --no-warnings --loader=esbuild-register/loader --require=esbuild-register --input-type=module
/Users/futaiwata/nitrogql/examples/nextjs/src/schema/scalar.ts:1
import { GraphQLSchema } from "graphql";
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at internalCompileFunction (node:internal/vm:73:18)
    at loadCJSModule (node:internal/modules/esm/translators:155:23)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:234:7)
    at ModuleJob.run (node:internal/modules/esm/module_job:217:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
    at async loadSchemaJs (file:///Users/futaiwata/nitrogql/examples/nextjs/node_modules/@nitrogql/core/dist/loader.js:5:18)

Node.js v20.8.1

    at checkExecSyncError (node:child_process:890:11)
    at execFileSync (node:child_process:926:15)
    at executeNodeSync (file:///Users/futaiwata/nitrogql/examples/nextjs/node_modules/@nitrogql/core/dist/config.js:12:12)
    at execute_node (file:///Users/futaiwata/nitrogql/examples/nextjs/node_modules/@nitrogql/core/dist/config.js:47:24)
    at wasm://wasm/0081e93e:wasm-function[742]:0x1a620c
    at wasm://wasm/0081e93e:wasm-function[21]:0x7921
    at wasm://wasm/0081e93e:wasm-function[1677]:0x1d8d49
    at wasm://wasm/0081e93e:wasm-function[690]:0x19fc2f
    at wasm://wasm/0081e93e:wasm-function[1715]:0x1d95d6 {
  status: 1,
  signal: null,
  output: [
    null,
    '',
    '/Users/futaiwata/nitrogql/examples/nextjs/src/schema/scalar.ts:1\n' +
      'import { GraphQLSchema } from "graphql";\n' +
      '^^^^^^\n' +
      '\n' +
      'SyntaxError: Cannot use import statement outside a module\n' +
      '    at internalCompileFunction (node:internal/vm:73:18)\n' +
      '    at loadCJSModule (node:internal/modules/esm/translators:155:23)\n' +
      '    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:234:7)\n' +
      '    at ModuleJob.run (node:internal/modules/esm/module_job:217:25)\n' +
      '    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)\n' +
      '    at async loadSchemaJs (file:///Users/futaiwata/nitrogql/examples/nextjs/node_modules/@nitrogql/core/dist/loader.js:5:18)\n' +
      '\n' +
      'Node.js v20.8.1\n'
  ],
  pid: 70252,
  stdout: '',
  stderr: '/Users/futaiwata/nitrogql/examples/nextjs/src/schema/scalar.ts:1\n' +
    'import { GraphQLSchema } from "graphql";\n' +
    '^^^^^^\n' +
    '\n' +
    'SyntaxError: Cannot use import statement outside a module\n' +
    '    at internalCompileFunction (node:internal/vm:73:18)\n' +
    '    at loadCJSModule (node:internal/modules/esm/translators:155:23)\n' +
    '    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:234:7)\n' +
    '    at ModuleJob.run (node:internal/modules/esm/module_job:217:25)\n' +
    '    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)\n' +
    '    at async loadSchemaJs (file:///Users/futaiwata/nitrogql/examples/nextjs/node_modules/@nitrogql/core/dist/loader.js:5:18)\n' +
    '\n' +
    'Node.js v20.8.1\n'
}
Error:
Failed to execute config file: Failed to execute config file
npm ERR! Lifecycle script `generate` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: nextjs@1.3.0 
npm ERR!   at location: /Users/futaiwata/nitrogql/examples/nextjs 
```

</details>

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -30,11 +30,15 @@ jobs:
         run: cargo build --target wasm32-wasi
   unit-test-js:
     runs-on: ubuntu-latest
+    name: unit-test-js (Node ${{ matrix.node-version }})
+    strategy:
+      matrix:
+        node-version: [18, 20.5, 20]
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: 18
+          node-version: ${{ matrix.node-version }}
       - run: npm ci
       - run: npm run check --workspaces --if-present
       - run: npm test --workspaces --if-present
@@ -42,7 +46,9 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [18, 20]
+        # 20.5 is added because @nitrogql/esbuild-register behaves differently
+        # before and after 20.6
+        node-version: [18, 20.5, 20]
     name: e2e-test (Node ${{ matrix.node-version }})
     steps:
       - uses: actions/checkout@v3

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -43,6 +43,7 @@ jobs:
     strategy:
       matrix:
         node-version: [18, 20]
+    name: e2e-test (Node ${{ matrix.node-version }})
     steps:
       - uses: actions/checkout@v3
       - name: install rustup

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -40,6 +40,9 @@ jobs:
       - run: npm test --workspaces --if-present
   e2e-test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [18, 20]
     steps:
       - uses: actions/checkout@v3
       - name: install rustup
@@ -49,7 +52,7 @@ jobs:
         run: rustup target add wasm32-wasi wasm32-unknown-unknown
       - uses: actions/setup-node@v3
         with:
-          node-version: 18
+          node-version: ${{ matrix.node-version }}
       - run: npm install
       - name: build rust
         run: ./build/build-rust.sh

